spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/XDParser.java

```diff
@@ -43,4 +45,29 @@
 	 *         described by the DSL.
 	 */
 	List<ModuleDescriptor> parse(String name, String config, ParsingContext type);
+
+	public static class DefinitionOrException {
```

Comment:
The problem is we want exceptions **per line** (_i.e._ in a multiline document such as
```
1: foo = http | log
2: bar = tap:stream:foo2 > log
```
we want to report an error at line 2 but still have a parse result for line 1)